libres/lib/python/logging.cpp

```diff
@@ -0,0 +1,134 @@
+#include <unordered_map>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <ert/python.hpp>
+
+using namespace std::string_literals;
+namespace py = pybind11;
+
+// Controls the prefix of the Python logger:
+//   logging.getLogger(f"{LOGGER_NAMESPACE}.{name}")
+#define LOGGER_NAMESPACE "res"s
+
+namespace {
+class Logger : public ert::ILogger {
+    struct interface {
+        py::object debug;
+        py::object info;
+        py::object warning;
+        py::object error;
+        py::object critical;
+
+        interface(py::object logger);
+    };
+
+    std::unique_ptr<interface> m_interface;
+
+public:
+    void init(const std::string &name);
+
+    /** Called on Python shutdown to release the py::objects */
+    void shutdown() { m_interface = nullptr; }
+
+protected:
+    void log(Level level, fmt::string_view f, fmt::format_args args) final;
+};
+
+bool has_init_logging{};
```

Comment:
Explicit init for clarity?

Reply:
It's so we know not to use Python functions while Python is not ready (`init_logging` has not been called yet)

Reply:
I meant to perhaps explicitly init it to `false` to be explicit  😄

Reply:
The `{}` means false. Do you want me to be even more explicit?

Reply:
@eivindjahren thoughts?

Reply:
It it's well-defined syntax then leave it! 😄  (Possibly add a comment.)

(Been some time since I [learned C](https://en.wikipedia.org/wiki/The_C_Programming_Language), then C++ when it came along and mainly Java and Python the last two decades. I'm rusty on syntax...  😬 )